_posts/2016-08-02-testing-spark-streaming-unit-testing.md

```diff
@@ -0,0 +1,126 @@
+---
+layout: post
+asset-type: post
+name: Testing-Spark-Streaming-Unit-Testing
+title: 'Testing Spark Streaming: Unit testing'
+date: 2016-08-02 00:20:00 +00:00
+author: Felipe Fernández
+image:
+    src: /assets/img/custom/blog/unit_testing.jpg
+    attribution:
+      text: Untitled by Daida Medina
+      href: https://www.instagram.com/p/BHiJPKrBb-w/?taken-by=thelastfreeusername
+tags:
+- testing
+- big data
+- spark
+- streaming
+
+---
+
+There is enough evidence to prove the importance of automated testing. Projects in new fields often neglect automated testing, as the domain itself steals the attention of the developers. However, lack of testing implies 'laugh now, cry later'. Some of the tools around Big Data space has been architected around testability or, at least, the community has taken care of it afterwards. We'll see how [Spark](http://spark.apache.org/), and more specifically [Spark Streaming](http://spark.apache.org/streaming/), performs in different aspects of automated testing.
+
+## What is Stream Processing
+
+[Stream processing](https://www.wikiwand.com/en/Stream_processing) is a programming paradigm that works on infinite and continuous streams of data, applying parallel operations on top. The idea is simple but powerful, and the complexity of the implementation will vary depending on the following requirements:
```

Comment:
parallel operations on top ==> parallel operations on `them` - is that what you mean to say?